File: leetcode/leetcode_cpp/maximum-average-subarray-i.cpp
/*
https://leetcode.com/problems/maximum-average-subarray-i

Solution1. Sliding Window
time: o(n)
space: o(1)
*/
class Solution {
public:
    double findMaxAverage(vector<int>& nums, int k) {
        int l = 0, r = 0;
        double sum = 0;
        while (r < k) sum += nums[r++];
        double res = sum;

        while (r < nums.size()) {
            sum += nums[r] - nums[l];
            res = max(res, sum);
            r += 1;
            l += 1;
        }
        return res / k;
    }
};

File: leetcode/leetcode_py/maximum-average-subarray-i.py
'''
https://leetcode.com/problems/maximum-average-subarray-i

Solution1. Sliding Window
time: o(n)
space: o(1)
'''
class Solution:
    def findMaxAverage(self, nums: List[int], k: int) -> float:
        s = 0.0
        l, r = 0, 0
        while r < k:
            s += nums[r]
            r += 1
        res = s
        while r < len(nums):
            s += nums[r] - nums[l]
            res = max(res, s)
            l += 1
            r += 1
        return res / k
            
